### PR TITLE
Upgrade `react native` from  `0.67.5` to `0.69.10`

### PR DESCRIPTION
**Pull request by siddarthkay**
### Summary
This PR upgrades react-native version from 0.67.5 to 0.69.10 and does what is necessary to make the upgrade happen.
This PR is yet again another stepping stone to latest version.
When this PR will be merged in we will be using a `react-native` version that is officially supported by Facebook's React Native team.

Journey to upgrade Android Side can be seen here : https://github.com/status-im/status-mobile/pull/15995
Journey to upgrade IOS Side can be seen here : https://github.com/status-im/status-mobile/pull/15645

status: ready

### CODE REVIEW

**Comment by J-Son89** [comment]
Awesome work @siddarthkay! Given the Porto UI fit & finish is next week I think we should avoid merging this until afterwards in case it disrupts the dev environments 👍

**Comment by siddarthkay** [comment]
@J-Son89 : Sure, I agree. 
At the moment the purpose of this PR is to get a proper QA done and then I'll wait for the upcoming release and the UI event.

**Review by alwx — APPROVED** [review]
Amazing job, @siddarthkay! 
I saw some hacks you did (like the one in [nix/deps/gradle/generate.sh](https://github.com/status-im/status-mobile/pull/16016/files#diff-3c2efcfb99106f065b63da031355b98d73137880bcd77f1ca14ea8b7e9683182)), I guess the only thing is to not forget to remove those hacks once we don't need them 😅

**Comment by siddarthkay** [comment]
Thanks @alwx ! yes for sure, Once https://github.com/status-im/status-mobile/issues/15447 gets fixed, I'll make sure to get rid of those hacks 😉

File: nix/deps/gradle/generate.sh
Comment (jakubgs) [inline nix/deps/gradle/generate.sh:89]: Yes, we will definitely need a proper fix for this, and theoretically that will be also a solution to this:

* https://github.com/status-im/status-mobile/issues/15447

File: ci/Jenkinsfile.ios
Comment (jakubgs) [inline ci/Jenkinsfile.ios]: Not sure what those formatting changes are for. Please revert them.

File: android/app/src/main/java/im/status/ethereum/MainActivity.java
Comment (jakubgs) [inline android/app/src/main/java/im/status/ethereum/MainActivity.java]: Why is this here and commented out?

File: Makefile
Comment (jakubgs) [inline Makefile]: What is this for? I assume debugging. Then we don't need this.

File: nix/deps/gradle/gradle_parser.awk
Comment (jakubgs) [inline nix/deps/gradle/gradle_parser.awk]: Maybe we should be a bit more explicit?
```suggestion
        if (line ~ "com.facebook.react:(react-native|hermes-engine)") { continue }
```

File: ios/Podfile
Comment (jakubgs) [inline ios/Podfile]: Is this line necesary?

File: ios/Podfile
Comment (jakubgs) [inline ios/Podfile:32]: If we are putting the closing bracket on the same indent level as function call the arguments should have a normal 2 space indent:
https://github.com/rubocop/ruby-style-guide#no-double-indent

**Review by jakubgs — CHANGES_REQUESTED** [review]
Good stuff.

File: ios/Podfile
Comment (siddarthkay) [reply ios/Podfile]: nope, it was supposed to be part of the upgrade but the builds would fail and the upgrade works well without it..

File: nix/deps/gradle/gradle_parser.awk
Comment (siddarthkay) [reply nix/deps/gradle/gradle_parser.awk]: sounds good!

File: Makefile
Comment (siddarthkay) [reply Makefile]: yeah I wanted a make command to build PR flavours locally...but we don't need it that often so safe to remove.

File: android/app/src/main/java/im/status/ethereum/MainActivity.java
Comment (siddarthkay) [reply android/app/src/main/java/im/status/ethereum/MainActivity.java]: removed.

**Review by ajayesivan — COMMENTED** [review]
iOS is working fine 🥳 

On Android, I'm getting the following error when running locally,
```
error: builder for '/nix/store/1psjkjxd5c6icyqh4mwzfr081239z8n5-aapt2-7.3.0-8691043.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/gjfsvdrwb1yp0bignnpg0wdv02zxjy2l-status-mobile-maven-deps.drv' failed to build
error: build of '/nix/store/gjfsvdrwb1yp0bignnpg0wdv02zxjy2l-status-mobile-maven-deps.drv', '/nix/store/hxy6l1vz9av572hi27slv50yz258wary-status-go-0.151.15-7586452-android.drv' failed
make: *** [run-android] Error 1
```

**Comment by ajayesivan** [comment]
> iOS is working fine 🥳
> 
> On Android, I'm getting the following error when running locally,
> 
> ```
> error: builder for '/nix/store/1psjkjxd5c6icyqh4mwzfr081239z8n5-aapt2-7.3.0-8691043.drv' failed with exit code 2
> error: 1 dependencies of derivation '/nix/store/gjfsvdrwb1yp0bignnpg0wdv02zxjy2l-status-mobile-maven-deps.drv' failed to build
> error: build of '/nix/store/gjfsvdrwb1yp0bignnpg0wdv02zxjy2l-status-mobile-maven-deps.drv', '/nix/store/hxy6l1vz9av572hi27slv50yz258wary-status-go-0.151.15-7586452-android.drv' failed
> make: *** [run-android] Error 1
> ```

Fixed!

**Review by ajayesivan — APPROVED** [review]
@siddarthkay Awesome work 🎉

**Review by jakubgs — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @siddarthkay thank you for PR.
## ISSUE 1: App crashes upon launch on Android

## Devices:
- Real device: Huawei p20 light, android 9
- Android Studio Emulator: Pixel XL API 29

**logs:**
[android_logcat.txt](https://github.com/status-im/status-mobile/files/11591446/android_logcat.txt)

**Review by ilmotta — APPROVED** [review]

**Comment by siddarthkay** [comment]
The release android build insta crashing is something we saw in the previous upgrade as well : https://github.com/status-im/status-mobile/pull/15486#issuecomment-1500232783 
I'm investigating further.

File: android/app/build.gradle
Comment (OmarBasem) [inline android/app/build.gradle]: Why are we disabling hermes?

Comment (siddarthkay) [reply android/app/build.gradle]: We're not I'm just disabling on CI to see if android release builds work. my local linux OS environment is broken atm.
Most likely the last commit will be reverted..

**Comment by siddarthkay** [comment]
@VladimrLitvinenko : Issue 1 is fixed and the PR build no longer crashes :)

**Comment by VolodLytvynenko** [comment]
Hi @siddarthkay, thank you for the PR. The manual testing process has been going well so far.

However, we are encountering an issue on Android 10 devices which causes around 90% of the test cases to fail

## ISSUE 1: [Android 10] When launching the app, an error `java.langRuntimeException: Could not invoke WwebSocketModule.connet` occurs

### Steps:
https://app.eu-central-1.saucelabs.com/tests/bf6e5a88e7ab40b4b4078a7000dc9226?auth=ec438fed7aa498467a060d073f0642cf#2

### Actual result:
"java.langRuntimeException: Could not invoke WwebSocketModule.connet" error is shown during app launch
image

### Device:
- Real device: Huawei P40 lite, Android 10
- Sauce lab emulator: Pixel XL, Android 10

### Logs:
[reactupgrade.log](https://github.com/status-im/status-mobile/files/11651191/reactupgrade.log)

**Comment by siddarthkay** [comment]
Hi @VladimrLitvinenko : Thanks for taking the time to test and find out this issue, I do see this issue reported in official react-native repo as well and unfortunately none of them have valid solutions that we could use atm. 

ref : 
- https://github.com/facebook/react-native/issues/26007
- https://github.com/facebook/react-native/issues/37330
- https://github.com/facebook/react-native/issues/33065

I'll try to reproduce locally and investigate further 👍🏻

**Comment by VolodLytvynenko** [comment]
> Hi @VladimrLitvinenko : Thanks for taking the time to test and find out this issue, I do see this issue reported in official react-native repo as well and unfortunately none of them have valid solutions that we could use atm.

Got it. Thank you for the clarification. Unfortunately, if we merge this PR, it will result in the consistent failure of e2e tests in further PRs

**Review by rahulpsingh — APPROVED** [review]

**Comment by KassUA** [comment]
**ISSUE 2: App freezes on loading screen (opener with Status logo)**

How to reproduce: 
- Start the app / Force quit until Status logo freezes
- Background the app few times
- See if an error flashes in background

https://github.com/status-im/status-mobile/assets/97434915/8a2b1122-1d8f-497a-99fc-0889d2fb8453

 - Expand the app on full screen
 - Block the screen of the mobile
 - Unblock the screen of the mobile
 - See if the error message occurs
 
status error

**Info about device:**
Device name: Redmi Note 8 Pro
Android Version: 10 QP1A.190711.020
MIUI version: MIUI Global 12.0.8 Stable

**Comment by VolodLytvynenko** [comment]
Hi @siddarthkay, thank you for your work. No issues from my side. PR can be merged

**Pull request merged**
